### PR TITLE
Litmus concepts

### PR DESCRIPTION
**Pull request by DavidS**
new page with some background information

### CODE REVIEW

**Comment by DavidS** [comment]
This is still unfinished, but uploading it now to have a backup aside from my workstation and for others to have a peek

File: litmus-concepts.md [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+The following sections go over the various components, their reasoning and discuss alternative options.
+
+## UI
+
+The current UI/UX of Litmus is implemented as a set of rake tasks in [puppet_litmus:lib/puppet_litmus/rake_tasks.rb](https://github.com/puppetlabs/puppet_litmus/blob/master/lib/puppet_litmus/rake_tasks.rb).
+
+**Reasoning:**
+Rake is a ubiquitous choice of interacting with a set of connected tasks in the ruby world.
+The limited option parsing capabilities push the design towards simple interactions and storing important information in configuration files.
+It is very easy to get started with a project based on rake tasks, as there is a wealth of examples and titorials how to build rake tasks.
```
Comment (carabasdaniel) [inline litmus-concepts.md:57]:
```suggestion
It is very easy to get started with a project based on rake tasks, as there is a wealth of examples and tutorials on how to build rake tasks.
```

File: litmus-concepts.md [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+This allows users to re-use skills acquired and strenghtens Puppet's better-together story.
+
+**Alternatives:**
+Bolt as communication layer touches most other components and - while having a well-defined interface - would not be easy to replace.
+There is currently no reason to consider alternatives to Bolt.
+
+## Configuration
+
+Configuration for Litmus currently is spread over several files.
+The `provision.yml` contains various lists of platforms to target for tests.
```
Comment (carabasdaniel) [inline litmus-concepts.md:82]:
```suggestion
The `provision.yaml` contains various lists of platforms to target for tests.
```

File: litmus-concepts.md [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+The CI job setup (usually in `.travis.yml`, `appveyor.yml` or similar) contains the overall sequencing of steps for testing.
+This setup is usually the same everywhere and is encoded in pdk-templates.
+There are sometimes slight variation to choose the puppet version and platform combinations from `provision.yml`.
```
Comment (carabasdaniel) [inline litmus-concepts.md:85]:
```suggestion
There are slight variations to choose the puppet version and platform combinations from `provision.yaml`.
```

File: litmus-concepts.md [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+thus common public images are usually not representative of production on full VMs.
+To avoid this limitation, [puppetlabs/litmusimages](https://github.com/puppetlabs/litmusimage) provides a set of pre-fabbed [docker images](https://hub.docker.com/u/litmusimage) that allow SSH access and have an init process running.
+
+In some cases, using bare metal servers or already running systems is unavoidable.
+
+### How to provision Test Infrastructure
+
+There are as many ways to aqcuire access to test systems as there are kinds of test systems.
+By default, Litmus calls out to a provisioner task to provision or tear down systems on demand.
+In the `puppetlabs-provision` module, we ship a number of commonly needed provisioners for docker (with SSH), docker_exp (using bolt's docker transport), vagrant (for local VMs) and the private cloud APIs [VMpooler](https://github.com/puppetlabs/vmpooler) and [ABS](https://github.com/puppetlabs/always-be-scheduling).
```
Comment (carabasdaniel) [inline litmus-concepts.md:142]: ABS is a private repository with a private tool designed for this provisioning, not sure if it's necessary to mention it

File: litmus-concepts.md [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+
+Since Litmus 0.18 the rake tasks also allow calling arbitrary tasks outside the `provision` module to provision test systems. [Daniel's terraform demo](https://youtu.be/8BMo9DcZ4-Q) shows one application of this.
+
+The provision task will allocate/provision/start VMs or containers of the requested platforms and add them to the inventory file.
+
+Through the use of the inventory file Litmus can also consume arbitrary other systems by users supplying their own inventory, independently of Litmus' provision capabilities.
```
Comment (carabasdaniel) [inline litmus-concepts.md:148]:
```suggestion
Through the use of the inventory file Litmus can also consume arbitrary other systems by users supplying their own data, independently of Litmus' provision capabilities.
```

File: litmus-concepts.md (hunk 1 above)
Comment (DavidS) [reply litmus-concepts.md:57]: oops! :shame:

File: litmus-concepts.md (hunk 4 above)
Comment (DavidS) [reply litmus-concepts.md:142]:
```suggestion
In the `puppetlabs-provision` module, we ship a number of commonly needed provisioners for docker (with SSH), docker_exp (using bolt's docker transport), vagrant (for local VMs) and the private cloud APIs [VMpooler](https://github.com/puppetlabs/vmpooler) and the puppet-private [ABS](https://github.com/puppetlabs/always-be-scheduling).
```

Comment (DavidS) [reply litmus-concepts.md:142]: I'll leave it in but mention that it is puppet-private.

**Pull request merged**